### ISSUE TITLE
DenseNet example

### ISSUE THREAD

**Issue by brendanhasz** [brendanhasz/probflow#33]
Add an example of building a net using DenseNet blocks?  Something like

```python
class DenseNetBlock(pf.Module):
    
    def __init__(self, dims):
        self.num_layers = len(dims) - 2
        cdims = np.cumsum(dims[:-1])
        self.layers = [pf.Dense(cdims[i], cdims[i+1]) for i in range(self.num_layers)]
        self.batch_norms = [pf.BatchNormalization(d) for d in cdims[1:]]
        self.linear = pf.Dense(cdims[-1], dims[-1])  # last layer reduces
        
    def __call__(self, x):
        outputs = [x]
        for i in range(self.num_layers):
            x = tf.concat(outputs, -1)
            x = self.layers[i](x)
            x = tf.keras.activations.swish(x)
            x = self.batch_norms[i](x)
            outputs.append(x)
        return self.linear(tf.concat(outputs, -1))

        

class DenseNetNetwork(pf.Module):

    def __init__(self, dims):
        self.num_blocks = len(dims)
        self.blocks = [DenseNetBlock(d) for d in dims]
        self.batch_norms = [pf.BatchNormalization(d[-1]) for d in dims[:-1]]

    def __call__(self, x):
        for i in range(self.num_blocks):
            x = self.blocks[i](x)
            if i < self.num_blocks - 1:
                x = tf.keras.activations.swish(x)
                x = self.batch_norms[i](x)
        return x



class DenseNetRegression(pf.Model):
    
    def __init__(self, dims):
        assert dims[-1][-1] == 2
        self.net = DenseNetNetwork(dims)

    def __call__(self, x):
        x = self.net(x)
        return pf.Normal(x[..., 0], tf.exp(x[..., 1]))



model = DenseNetRegression([[256, 128, 64, 32, 16], [16, 128, 64, 2]])
```

Maybe even add `DenseNet` and `DenseNetNetwork` to modules, and `DenseNetRegression` to applications (though versions with more options (i.e. pass choice of activation function, whether or not you want batch normalization, etc).
